### ISSUE TITLE
:red_circle: VFS Product Directory Changes

### ISSUE THREAD

**Issue by dmkosko** [department-of-veterans-affairs/va.gov-team#52276]
What
We're looking to provide all individuals associated or interested as well as stakeholders involved with product this information. We feel this "one-stop shop" directory will allow for greater oversight of what each portfolio is maintaining, a reference guide for new product hires/onboarding, understanding if a Product Lead/Manager has too many things on their plate, and aligning priorities on the current and upcoming work. We envision the tweaking of the directory will allow for a more user friendly interface, providing the most beneficial information up front. A refresh of the product list will provide the most accurate, up to date repository of products identified from recent conversations with the crews.

Why
We feel making changes to this list will act as the "front door" for viewing products that the teams are working on. An updated list serves everyone in at least two ways, firstly awareness. Chris, decision makers under him, and stakeholders that rely on/help fund these products or interested parties are able to see where the work stands. Secondly, for individual/team health. We want to make sure one person/team does not have "too much" on their plate. Being able to filter some of these fields will give leadership the ability to understand who may be overworked and even help discuss how many products a team should be working on. We feel these pieces benefit numerous groups and has the potential to be more than just a product list.

Things to do

- [ ] Meet with Joe Tice to discuss changes we're proposing and LoE involved
- [ ] Discuss best approach to add in non-VA.gov items (meeting with technical team to discuss manifest file)
- [ ] Understand if the collab cycle request ticket should be updated to include a question about a product being associated with VA.gov or not (We may need to wait until we get feedback from the task above)
- [ ] Determine best approach to add/remove future products to the list
- [ ] Determine if the VFS Product Directory serves the needs of what we're trying to do

**Comment by dmkosko** [comment]
Link to the meetings minutes are [here](https://dvagov-my.sharepoint.com/:w:/r/personal/lisa_koenigsberg_va_gov/_layouts/15/Doc.aspx?sourcedoc=%7B27751B08-B16B-4C43-A392-FFAEC8BB3034%7D&file=VFSProdDirEnhancementMeeting_Jan192023.docx&action=default&mobileredirect=true&cid=70a805ac-e01b-4fcd-92f5-c1dbd1bb849d) 

- [ ] Joe to provide updates on when the product list view can be implemented by January 30th
- [ ] Dan to discuss with Joe when the sortable/filterable views can be implemented by January 30th
- [ ] Dan to set up a meeting with Joe, Lisa, Peter, and Kurt to discuss how the manifest file is created and non-VA.gov products by February 3rd
- [ ] Joe to investigate whether a question should be added to the collab cycle ticket to identify if a product is VA.gov or not by February 3rd

**Comment by dmkosko** [comment]
For the implementation work, Joe confirmed nothing is standing in our way from a technology perspective. He is working with his management to get these items on the schedule.

**Comment by dmkosko** [comment]
Meeting minutes are [here](https://dvagov-my.sharepoint.com/:w:/r/personal/lisa_koenigsberg_va_gov/_layouts/15/Doc.aspx?sourcedoc=%7B27751B08-B16B-4C43-A392-FFAEC8BB3034%7D&file=VFSProdDirEnhancementMeeting_Jan192023.docx&action=default&mobileredirect=true&cid=70a805ac-e01b-4fcd-92f5-c1dbd1bb849d)

**Comment by dmkosko** [comment]
Spoke to Joe and updates on implementation of the items are below. Tickets have been created for all three items.  

Product List View & Sortable/Filterable Fields - Will be implemented before the end of February 2023. Additional minor questions will be asked by Joe to Lisa and Dan prior to these items launching. 
Reordering of the fields in the Product Name View - Will be implemented before the end of February 2023. Even though we put this as a backlog item, Joe created the ticket so there is a record of it.

**Comment by dmkosko** [comment]
For the discussion about the manifest file and adding non-VA.gov products to the VFS Directory, there is some feedback from Joe below. Dan to discuss with Lisa if we will want to proceed with this meeting. 

Wanted to ensure we're on the same page with the goal of this discussion - My feeling is the main thing our team wants to understand is the need and use case for the list that you're trying to create...
From our perspective, and I suspect that I speak for both my team and platform when I say, we don't feel like it would be a good approach to add manifest files/directories for products that are not veteran-facing-products that are currently live on the site. The way we understand it currently is that the desire to do this is to get to one unified list of all types of products across the VA's enterprise, and the VFS Product Directory seems like a logical choice for that...
However - We believe doing this may have unwanted second & third order effects that would cause problems down the line with the way the VFS Product Directory is being used - both now and in the future...
I didn't want to take us down a rabbit hole in our last discussion into the details of since we were already over time - but can expand on this in our next discussion if necessary.  For now - it's helpful to know/understand, that with the VFS Product Directory in place and operational, it unblocks other efforts which are slowly progressing behind the scenes and we believe will add value to platform offerings... Adding non-VFS/VA-gov products to the VFS Product Directory that don't belong there would add unwanted noise to that functionality that would likely complicate if not derail those efforts.
We do however fully understand the need for an overarching list that has a broader view of things - To use military analogies, If we were to consider the VFS Product Directory as a 10,000 foot view, we understand the need for a 30,000 foot view. Where everything from the 10k view is in there, as well as much more - but the 10k and 30k views have different purposes and use cases driving them...
So that's where our thinking is in saying - we think the goal of the meeting is really focusing on what the need is for the 30k view, and look at how our 10k foot view can support and integrate with it, without risking compromising the integrity of the 10k view/data
The scope of a solution for this may very likely have a longer tail than the other smaller tasks, so I want to set initial expectations appropriately

**Comment by dmkosko** [comment]
After speaking to Lisa, we are going to hold off on this meeting. We don't want to mess with the manifest files and use it for anything that it was not intended for. I reached out to Joe asking if we would we be able to put a link on the VFS Directory for someone to find the list of non-VA.gov products and have them directed to a page of ours in GitHub. We would follow the same format as the VFS Directory and just waiting on a response from Joe to see if it's feasible.

**Comment by dmkosko** [comment]
From Joe today: the first of the other improvements - adding additional data to the VFS Product Directory listing page is underway this sprint.

**Comment by dmkosko** [comment]
Consolidated list of what remains:

- [ ] Addition/removal of products based on our spreadsheet
- [ ] Add the Crew into a sortable column
- [ ] Adding link from our product GH page to the VFS directory for non-VA.gov products
- [ ] Editing the summary on the home page of the directory
- [ ] Filling in any N/A's between points of contact and product outlines

**Comment by dmkosko** [comment]
Spoke to Joe Tice today. There were some items that needed to be elevated so a portion of the team has been working on that. I will continue staying close to Joe and I'm hoping things don't slow for too long. I am going to check in with him early next week and continue driving what we can at this time.

**Comment by dmkosko** [comment]
Asked product ops group to provide feedback on draft summary paragraph for VFS Directory by the end of this week. 

https://dsva.slack.com/archives/C03H546LHDY/p1678730920428529

**Comment by dmkosko** [comment]
The updated summary paragraph will be changed in production early next week per Joe Tice.

The item related to the removal of products on the VFS Directory is going into the next sprint (3/16 - 3/28) and that change being implemented will depend on how complex the answer is.

**Comment by dmkosko** [comment]
Link is fixed from the note above.

**Comment by dmkosko** [comment]
Can we leverage the Product Root URL field to add a link to the actual product on VA.gov. Either as another column on the table or as an additional <link> in the product name field. Waiting on more info from Joe

**Comment by dmkosko** [comment]
Joe will be creating a ticket to add a link from non-VA.gov products wto the VFS Product Directory

Regarding putting a link to the actual product on VA.gov, Joe's team blocked out the details of it in a text document, but only committed the new ticket today as they're logging the stuff that came up in the span of this past sprint...

**Comment by dmkosko** [comment]
Samara needs 2 products added that were previously seen as features. Link to the discussion is [here](https://dsva.slack.com/archives/CBU0KDSB1/p1680098098808479)

**Comment by dmkosko** [comment]
Sent Joe over the list of features and what products they roll up to. He is going to look at today and let me know if there are any questions.

**Comment by dmkosko** [comment]
Spoke to Joe and he provided an update on some of the items related to the VFS Product Directory. In order to remove some of the existing items identified as products, his team needs to modify the automated collection that's being done, and the mechanism that conducts it. They're prioritized that work for this sprint, with the plan that they'll update the VFS Product List after the automated scan has been removed from the process. 

I'm waiting to hear back on when the sprint is expected to finish up and where the other items are in the list.

**Comment by dmkosko** [comment]
Meeting minutes shared in the #product slack channel and attached [here](https://dvagov-my.sharepoint.com/:w:/r/personal/lisa_koenigsberg_va_gov/Documents/ProductOperations/[VFSDirectoryUpdatesMM_June72023.docx](https://dvagov-my.sharepoint.com/:w:/r/personal/lisa_koenigsberg_va_gov/Documents/ProductOperations/VFSDirectoryUpdatesMM_June72023.docx?d=we8cfd21ae0874a21a6386bba1bf1c93e&csf=1&web=1&e=z6PsmY)?d=we8cfd21ae0874a21a6386bba1bf1c93e&csf=1&web=1&e=z6PsmY). 

ACTION: While Joe and Curt are working through the removal process, Dan will sort the "NAs" on the VFS Directory and provide Joe and Curt the information that should be updated for the items staying on the list by June 26th 
ACTION: Curt and Joe are working through decoupling the product directory. There are some higher priority items on the docket currently, but once they are close to completed this item will be worked. Tentative completion date of June 30th

**Comment by dmkosko** [comment]
Spoke to Curt and he is going to incorporate a note in the "Product Link" field. Some product links require authentication to view and this change will inform visitors to the VFS Product Directory of that. Dan will circle back with Curt on an estiamted timeframe of when this will be going in.  

Side note, he is still working on the de-coupling of products.

**Comment by dmkosko** [comment]
Spoke to Curt and the de-coupling of the product directory from the manifest files is complete, waiting on code approval which is expected to come EOD today. Will coordinate next steps with Joe.

**Comment by dmkosko** [comment]
Joe and Curt are in the process of removing the products from our discussions. There are 40 items on the list and they got through 33 so far. I owe them further clarification on 7 items. 

Drafted a communication which will go out after Lisa's review.

**Comment by dmkosko** [comment]
This week, sent updates on the remaining 7 items to Joe and Curt (waiting for any questions). Will follow up next week to determine if they've been removed. 

Waiting on one final VFS PM name to come back for Veteran ID Cards and then will send them over the missing "NA" info

**Comment by dmkosko** [comment]
Sent over the "NA" information to Curt and Joe and waiting to hear when the changes will be made

**Comment by dmkosko** [comment]
https://github.com/department-of-veterans-affairs/va.gov-team/issues/45134#issuecomment-1779554832
